src/scripts/sniperino.ts

```diff
@@ -0,0 +1,139 @@
+    }
+
+    // Get a sniper with a given name. If one doesn't exist, create it.
+    let getOrCreateSniper = (name: string) => {
+    
+        // Check for active sniper, if not, create one
+        if (robot.brain.get("sniperino")[name] == null) {
+            let s = robot.brain.get("sniperino");
+            s[name] = new Sniper(name);
+            robot.brain.set("sniperino", s);
+        }
+        return robot.brain.get("sniperino")[name];
+    }
+
+    // Just get a sniper. Return null on failure.
+    let getSniper = (name: string) => {
+        return robot.brain.get("sniperino")[name];
+    }
+
+    // Respond to a stats message
+    robot.respond(/stats( me)?/i, (res) => {
+
+        let snipers = robot.brain.get("sniperino");
+        let retStr = "";
+
+        // Create a stats dictionary
+        for (var s in snipers) {
+            retStr += `${s} : ${snipers[s].getWinRate()}%\n`;
+        }
+        res.send(retStr);
+
+        // TODO: Sort these by best score
+    });
+
+
+
+    // Respond to a sniperino message
+    robot.respond(/sniperino( me)?/i, (res) => {
+
+        // Define instance of our Sniper object from getSniper method
+        let name = res.message.user.name;
+        let sniper = getOrCreateSniper(name);
+
+        // Deny the play if someone is already actively sniping. Otheriwise, display their roll.
+        if (sniper.currentSnipe == undefined) {
+            
+            // Player isn't playing, start a game for them
+            sniper.generateNewSnipe();
+        
+            // Send their game information
+            res.send(`ヽ༼ຈل͜ຈ༽_•︻̷̿┻̿═━一 ༼ つ ಥ_ಥ ༽つ${sniper.name}, roll higher than a ${sniper.currentSnipe} or the donger gets it!`);    
+            
+            // Save this active game back to the brain
+            robot.brain.set(name, sniper);
+        }
+        else {
+            // Player is already playing, yell at them because they are an idiot
+            res.send(`(ノಠ益ಠ)ノ彡┻━┻ YOU ARE ALREADY PLAYING SNIPERINO, ${sniper.name}. I oughtta sniperino YOU! ༼ຈل͜ຈ༽_•︻̷̿┻̿═━一`);
+        }
+    });
+
+    robot.on("roll", (res, roll) => {
+        
+        // Get our currently rolling sniper
+        let sniper = getSniper(res.message.user.name);
+        
+        if (sniper != null && sniper.currentSnipe > 0) {
+           
+            if (roll > sniper.currentSnipe) {
+                res.send(`(◠‿◠✿) ${sniper.name}, you roll a ${roll} and the donger lives! The donger thanks you (◠‿◠✿)`);
+                sniper.gamesWon += 1;
+            }
+            else if (roll == sniper.currentSnipe) {
+                res.send(`ヽ༼ຈل͜ຈ༽/ ${sniper.name}, you roll a ${roll} and tie! The donger is merely wounded. He will recover! ヽ༼ຈل͜ຈ༽/`);
+            }
+            else {
+                res.send(`༼ つ x_x ༽つThe donger is dead. ${sniper.name}, you did this! You MONSTER! ༼ つ x_x ༽ つ`);
+            }
+         
+        sniper.gamesPlayed += 1;
+        sniper.currentSnipe = undefined; 
```

Comment:
I don't particularly like the fact that (currentSnipe == null) is our check for whether a user is currently in the middle of a game or not. I feel like there's a better way. But this will work for now